src/main/core/work/task.rs

```diff
@@ -197,20 +172,50 @@ pub mod export {
         Box::into_raw(Box::new(task))
     }
 
+    /// Create a new reference-counted task for the current Host.
+    ///
+    /// SAFETY: see `taskref_new_for_host`
+    #[no_mangle]
+    pub unsafe extern "C" fn taskref_new(
+        callback: TaskCallbackFunc,
+        object: *mut libc::c_void,
+        argument: *mut libc::c_void,
+        object_free: TaskObjectFreeFunc,
+        argument_free: TaskArgumentFreeFunc,
+    ) -> *mut TaskRef {
+        let host_id = Worker::with_active_host_info(|i| i.id);
+        unsafe {
+            taskref_new_for_host(
+                host_id.unwrap(),
+                callback,
+                object,
+                argument,
+                object_free,
+                argument_free,
+            )
+        }
+    }
+
     /// Creates a new reference to the `Task`.
+    ///
+    /// SAFETY: `task` must be a valid pointer.
     #[no_mangle]
     pub unsafe extern "C" fn taskref_clone(task: *const TaskRef) -> *mut TaskRef {
         let task = unsafe { task.as_ref() }.unwrap();
         Box::into_raw(Box::new(task.clone()))
     }
 
     /// Destroys this reference to the `Task`, dropping the `Task` if no references remain.
+    ///
+    /// SAFETY: Host lock for the TaskRef's `object` and `argument` must be held.
     #[no_mangle]
     pub unsafe extern "C" fn taskref_drop(task: *mut TaskRef) {
         unsafe { Box::from_raw(task) };
     }
```

Comment:
There should be a null pointer check somewhere I think. (Or we could specify in "SAFETY" that the ptr must be valid, but I think a check would be better.)

Reply:
Done